### PR TITLE
Add NANO step and Mini+Nano validations to FastSim workflows

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:

This PR updates the FastSim steps in `runTheMatrix` following the presentation at the [SIM meeting on July 15](https://indico.cern.ch/event/1182398/) by
- Adding NANO step
   - By doing this, we need to split GEN step out from the main FastSim sequence. This PR then allows to reuse GEN datasets in future for FastSim validation (few lines of code needed to be enable when datasets are available). 
- Adding MiniAOD validation (as in https://github.com/cms-sw/cmssw/pull/38736, closed)
   - Few modifications on POG validation sequences are needed to support FastSim without HLT
- Adding NanoAOD validation
- Updating harvesting step to get less complain on missing histograms in FastSim

Note that this PR can take over https://github.com/cms-sw/cmssw/pull/38736

Examples of DQM from this PR: https://tinyurl.com/2yb7zl22
- MiniAOD and NanoAOD folders appears as expected. However, POG experts should check if something is missing.
- Compare to https://tinyurl.com/2ykx26v2 which come without this PR.

Note:
- There is a warning from JetMET analyzer as it looks for HLT. I assume this is a task of JetMET to modify the code to accept NoHLT from FastSim.
```
%MSG-e HLTConfigProvider:   JetAnalyzer:jetDQMAnalyzerAk4PFCHSPuppiCleanedMiniAOD@streamBeginRun  23-Jul-2022 14:31:36 CEST Run: 1 Stream: 0
Falling back to ProcessName-only init using ProcessName 'HLT' !
```
- This PR removes the following sequences from Tau validation sequence when `fastsim` era is used, to support FastSim/MC. This should be reviewed by Tau expert.
```
tauValidationMiniAODRealData,tauValidationMiniAODRealElectronsData,tauValidationMiniAODRealMuonsData
produceDenominatorRealData,produceDenominatorRealElectronsData,produceDenominatorRealMuonsData
```
- The follow issue should be followed up, to review how to use validation and dqm sequences in MC relvals, https://github.com/cms-sw/cmssw/issues/38828

#### PR validation:
- Run the following configs our from `runTheMatrix` give expected results:
`runTheMatrix.py --what upgrade -l 13234.0,13434.0,13240.303  -t 8 --wm init`
- DQM harvesting output is uploaded to GUI, to check.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
No need of backport.

### CODE REVIEW

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
assign fastsim

**Comment by srimanob** [comment]
Hi @cms-sw/fastsim-l2 @cms-sw/pdmv-l2 @cms-sw/dqm-l2 
Sorry for too many updates in past week, all I need is done in this PR. Could you please start to review? 
Note that, I am not sure why the last PR test can't get report on additional histograms on mini and nano. With this PR, I can get histograms as expected. I will re-trigger the test again.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by emanueleusai** [comment]
+1

**Comment by srimanob** [comment]
Ping @cms-sw/pdmv-l2 @cms-sw/fastsim-l2 
Thanks

**Comment by srimanob** [comment]
+Upgrade

**Comment by kskovpen** [comment]
+pdmv

**Comment by srimanob** [comment]
@cmsbuild please test

Just to refresh the test.

**Comment by civanch** [comment]
+1

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -148,6 +148,8 @@ def condition(self, fragment, stepList, key, hasHarvest):
     def preventReuse(self, stepName, stepDict, k):
         if "Sim" in stepName:
             stepDict[stepName][k] = None
+        if "Gen" in stepName:
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:151]: `elif` would have been more appropriate, but it doesn't really matter here

**Comment by perrotta** [comment]
> * This PR removes the following sequences from Tau validation sequence, to supprot FastSim/MC. This should be reviewed by Tau expert.
> 
> ```
> tauValidationMiniAODRealData,tauValidationMiniAODRealElectronsData,tauValidationMiniAODRealMuonsData
> produceDenominatorRealData,produceDenominatorRealElectronsData,produceDenominatorRealMuonsData
> ```

Could @cms-sw/tau-pog-l2 either approve or comment about the removal of the Tau validation sequences in FastSim?

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:151]: Thanks @perrotta 
I first use `elif` but see some unexpected issue. I can investigate more in the next PR, already plan to update the prod-like workflow for FastSim, and this point can be checked with detail.

**Comment by srimanob** [comment]
> > * This PR removes the following sequences from Tau validation sequence, to supprot FastSim/MC. This should be reviewed by Tau expert.
> > 
> > ```
> > tauValidationMiniAODRealData,tauValidationMiniAODRealElectronsData,tauValidationMiniAODRealMuonsData
> > produceDenominatorRealData,produceDenominatorRealElectronsData,produceDenominatorRealMuonsData
> > ```
> 
> Could @cms-sw/tau-pog-l2 either approve or comment about the removal of the Tau validation sequences in FastSim?

I realize that I did not make it clear from the original PR description. The removal is done only when `fastsim` era is used. So, there will be no effect on FullSim or Data validations. Good to hear from @cms-sw/tau-pog-l2

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:151]: Wrong comment before, deleted. Yes, this can change to `elif`.

**Comment by danielwinterbottom** [comment]
Hi, from the TauPOG side we are happy with removing these 6 sequences for FastSim - the sequences in question are intended only for data validation and therefore not needed for FastSim

**Comment by qliphy** [comment]
+1

**Comment by srimanob** [comment]
> Hi, from the TauPOG side we are happy with removing these 6 sequences for FastSim - the sequences in question are intended only for data validation and therefore not needed for FastSim

Thanks for the comment @danielwinterbottom 
In fact, the sequences we are talking about seems to run also on MC FullSim RelVals. I have opened an issue to follow up at https://github.com/cms-sw/cmssw/issues/38828.

**Pull request merged**